### ISSUE TITLE
replace(::String, count=1) with empty string regression

### ISSUE THREAD

**Issue by mortenpi** [JuliaLang/julia#36953]
On 1.4.2:

```
julia> replace("xyx", "" => '_', count=1)
"_xyx"
```

vs. 1.5.0 and master:

```
julia> replace("xyx", "" => '_', count=1)
"_x_y_x_"
```

**Comment by mortenpi** [comment]
Git bisect blames a6e27b38bdd16967c9675e4621805b8a3dc23d02 from #32348. I assume though that it's just revealing a bug in `replace`.

cc @sostock

**Comment by sostock** [comment]
The problem is this line:
https://github.com/JuliaLang/julia/blob/7c0cb30f5bc3bc2e0c520bcc302f713c5738de9a/base/strings/util.jl#L524
`r == 0:-1` was never true with the old behavior, but now `0:-1` is equal to any empty range. There was a similar bug with `split` (which was caught by the tests and thus fixed in #32348). I can prepare a fix.

**Comment by vtjnash** [comment]
This is a side-effect of the larger bug https://github.com/JuliaLang/julia/issues/36768
